src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1933,6 +1943,14 @@ bool InterpreterCreateQuery::doCreateTable(ASTCreateQuery & create,
             res->getName());
     }
 
+    if (!res->supportsDynamicSubcolumns() && hasDynamicSubcolumns(res->getInMemoryMetadataPtr()->getColumns()) && mode <= LoadingStrictnessLevel::CREATE)
+    {
+        throw Exception(ErrorCodes::ILLEGAL_COLUMN,
+            "Cannot create table with column of type Dynamic or JSON, "
+            "because storage {} doesn't support dynamic subcolumns",
```

Comment:
Looks like it broke a few tests 👀

Reply:
Fixed. It turned out we can create a MV with not existing destination table and check `StorageMaterializedView::supportsDynamicSubcolumns()` tries to call `supportsDynamicSubcolumns` from destination table that doesn't exists. Fixed it by checking if destination table exists in `StorageMaterializedView::supportsDynamicSubcolumns()` (we will check for dynamic subcolumns support during future destination table creation anyway)

Reply:
`03290_mix_engine_and_query_settings` fails though

Reply:
About `03290_mix_engine_and_query_settings`: Keep in mind that I used JSON just to check that you could use both table setting and query settings (`enable_json_type`) together. Feel free  to change it to something else, like suspicious codecs or similar.